src/Tribe/Tickets_Handler.php

```diff
@@ -1187,12 +1209,66 @@ public function get_ticket_max_purchase( $ticket_id ) {
 		 *
 		 * @since 4.8.1
 		 *
-		 * @param int                           $available Max purchase quantity.
-		 * @param Tribe__Tickets__Ticket_Object $ticket    Ticket object.
-		 * @param WP_Post                       $event     Event post.
-		 * @param int                           $ticket_id Raw ticket ID.
+		 * @param int                           $available_at_a_time Max purchase quantity, as restricted by Max At A Time.
+		 * @param Tribe__Tickets__Ticket_Object $ticket              Ticket object.
+		 * @param WP_Post                       $event               Event post.
+		 * @param int                           $ticket_id           Raw ticket ID.
+		 */
+		$available_at_a_time = apply_filters( 'tribe_tickets_get_ticket_max_purchase', $available_at_a_time, $ticket, $event, $ticket_id );
+
+		// Protect against filters passing `-1` as unlimited (from filters not yet updated for logic from version TBD).
+		if ( - 1 === $available_at_a_time ) {
+			$available_at_a_time = $max_at_a_time;
+		}
+
+		// If somehow oversold, set max allowed to zero.
+		if ( 0 > $available_at_a_time ) {
+			$available_at_a_time = 0;
+		}
+
+		return $available_at_a_time;
+	}
+
+	/**
+	 * Get the maximum quantity allowed to be added to cart in a single action, for performance and sanity reasons.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Ticket_Object $ticket Ticket object.
+	 *
+	 * @return int
+	 */
+	private function get_max_qty_allowed_to_be_added_to_cart_at_a_time( Tribe__Tickets__Ticket_Object $ticket ) {
+		$default_max = 100;
+
+		if ( ! $ticket instanceof Tribe__Tickets__Ticket_Object ) {
+			return $default_max;
+		}
+
+		/**
+		 * Cap the amount of tickets able to be purchased at a single time (single "add to cart" action)
+		 * for sanity and performance reasons.
+		 *
+		 * Anything less than `1` will be ignored and reset to the default.
+		 *
+		 * @since TBD
+		 *
+		 * @param int                           $default_max Maximum quantity allowed at one time (only applicable if
+		 *                                                   the ticket stock available is greater).
+		 * @param Tribe__Tickets__Ticket_Object $ticket      Ticket object.
+		 *
+		 * @return int
 		 */
-		return (int) apply_filters( 'tribe_tickets_get_ticket_max_purchase', $available, $ticket, $event, $ticket_id );
+		$max_at_a_time = absint(
+			apply_filters( 'tribe_tickets_get_ticket_default_max_purchase', $default_max, $ticket )
```

Comment:
Move filter above the return, since you return a default value, it should also be filterable on that case.

Reply:
👍